### PR TITLE
Add Graphql

### PR DESCRIPTION
**Pull request by mustaphaturhan**
Adds graphql to project. With this feature, we can use graphql in anywhere that we want.

Example usage:

1. Go to component that you want to add query or mutation or something,
2. Change `export default componentName` with `export default withApollo(componentName)` and do not forget to import { withApollo } from 'utils/apollo';
3. Write your query:
```
const GET_COMMENTS = gql`
  query {
    allComments {
      text
    }
  }
`;
```
4. Import { useQuery } from '@apollo/react-hooks';
5. Use `const { loading, error, data } = useQuery(GET_COMMENTS);` to send request
6. Enjoy.

### CODE REVIEW

File: packages/frontend/utils/apollo.js [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+
+/**
+ * Creates and provides the apolloContext
+ * to a next.js PageTree. Use it by wrapping
+ * your PageComponent via HOC pattern.
+ * @param {Function|Class} PageComponent
+ * @param {Object} [config]
+ * @param {Boolean} [config.ssr=true]
+ */
+export function withApollo(PageComponent, { ssr = true } = {}) {
```
Comment (msusur) [inline packages/frontend/utils/apollo.js]: Looking forward to seeing the tests for this 🕺

File: packages/frontend/utils/apollo.js [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+                  ...pageProps,
+                  apolloClient,
+                }}
+              />
+            );
+          } catch (error) {
+            // Prevent Apollo Client GraphQL errors from crashing SSR.
+            // Handle them in components via the data.error prop:
+            // https://www.apollographql.com/docs/react/api/react-apollo.html#graphql-query-data-error
+            console.error('Error while running `getDataFromTree`', error);
```
Comment (msusur) [inline packages/frontend/utils/apollo.js]: I'm a bit confused, where are we logging the error responses?

File: packages/frontend/utils/apollo.js (hunk 1 above)
Comment (mustaphaturhan) [reply packages/frontend/utils/apollo.js]: it is hard but i will try 😄

File: packages/frontend/utils/apollo.js (hunk 2 above)
Comment (mustaphaturhan) [reply packages/frontend/utils/apollo.js]: @msusur i have no idea but graphql is works as expected. official apollo/graphql documentation isn't working on ssr/nextjs. here is blog post about that: https://blog.apollographql.com/whats-next-js-for-apollo-e4dfe835d070

nextjs has an example that works with apollo but they have no explanation about how: https://github.com/zeit/next.js/tree/master/examples/with-apollo.

File: packages/frontend/utils/apollo.js [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+/**
+ * Creates and configures the ApolloClient
+ * @param  {Object} [initialState={}]
+ */
+function createApolloClient(initialState = {}) {
+  // Check out https://github.com/zeit/next.js/pull/4611 if you want to use the AWSAppSyncClient
+  return new ApolloClient({
+    ssrMode: typeof window === 'undefined', // Disables forceFetch on the server (so queries are only run once)
+    link: new HttpLink({
+      uri: 'https://api.graph.cool/simple/v1/ciyz901en4j590185wkmexyex', // Server URL (must be absolute)
```
Comment (mustaphaturhan) [inline packages/frontend/utils/apollo.js]: @msusur should I use our backend endpoint in here or should I define something to env files?

File: packages/frontend/utils/apollo.js (hunk 1 above)
Comment (mustaphaturhan) [reply packages/frontend/utils/apollo.js]: @msusur i added some tests but we are on lower coverage. for apollo, there is 50% statements, 60% branches, 80% functions and 50% lines covered.

**Pull request merged**